snapcraft/storeapi/__init__.py

```diff
@@ -762,16 +762,19 @@ def track(self):
 
         content = {}
         for indicator_count in itertools.count():
+            progress_indicator.update(indicator_count)
             if not queue.empty():
                 content = queue.get()
                 if isinstance(content, Exception):
                     raise content
-                widgets[0] = self._get_message(content)
-            progress_indicator.update(indicator_count)
             if content.get('processed'):
                 break
+            else:
+                widgets[0] = self._get_message(content)
             sleep(0.1)
         progress_indicator.finish()
+        # Print at the end to avoid a left over spinner artifact
+        print(self._get_message(content))
```

Comment:
I cry every time I see a print :'(
Let's make a sprint in Costa Rica to fix the UI and the logging!